src/rust/engine/src/context.rs

```diff
@@ -126,28 +127,27 @@ impl Core {
       })
       .unwrap_or_else(|e| panic!("Could not initialize Store: {:?}", e));
 
-    let underlying_command_runner: Box<dyn CommandRunner> = match &remote_execution_server {
-      Some(ref address) => Box::new(process_execution::remote::CommandRunner::new(
+    let command_runner = match &remote_execution_server {
+      Some(ref address) => BoundedCommandRunner::new(Box::new(process_execution::remote::CommandRunner::new(
         address,
         remote_execution_process_cache_namespace.clone(),
         remote_instance_name.clone(),
         root_ca_certs.clone(),
         oauth_bearer_token.clone(),
         remote_execution_extra_platform_properties.clone(),
+        // This param sets the grpc thread pool size. We use the local_execution_parallelism,
```

Comment:
I don't know whether this is a reasonable assumption, but happy to trust you've done enough research to validate it, and that we can change it in the future if needed :)

Reply:
It is only reasonable if you assume that users will be picking suitable values for the local workload, some balance between cpu and io bound. But in general I added the comment because it was confusing to me, and that was the only justification I could come up with.

Reply:
Mm, I think that I said something to this effect. I haven't looked into how this threadpool is used, ie: whether each ongoing requests blocks a thread for as long as the request is ongoing, or whether this is only where responses from the server are completed.

Answering that question would be necessary to choose this. @hrfuller : if you have time to investigate this and/or ask upstream before landing, that would be helpful. If you don't, you could safely leave this using the `remote_` number instead, with a TODO pointing to a question posed upstream to https://github.com/pingcap/grpc-rs about it.

Reply:
I can spend a bit of time trying to figure out the answer. I see your point that using the `remote_` number instead is the correct refactoring to maintain how it was before, so I will make that change if I don't uncover anything.

Reply:
What what I can tell looking at the grpc-rs code is that each thread polls for responses coming from Q's of completed requests. The requests are RR'd between all the Q's and the threads polling doesn't block outgoing requests, which happen synchronously. Also interestingly if you use a `grcpio::EnvBuilder` it configures the default thread count to equal the machines CPU count. I think we should stick close to that number, or just use the default.